cosmos/keybase.go

```diff
@@ -81,10 +85,25 @@ func (kb *Keybase) Delete(name, passphrase string, skipPass bool) error {
 }
 
 // Sign is a lock protected version of keys.Sign
+// it also keeps the last private key used in memory for the time set in `keepPrivTime` for performance improvement.
 func (kb *Keybase) Sign(name, passphrase string, msg []byte) ([]byte, crypto.PubKey, error) {
 	kb.mx.Lock()
```

Comment:
Shoudn't mutex only surround `kb.kb.ExportPrivateKeyObject`, should it?

I guess engine losses a lot of time while waiting for Sing  the transaction (and a bit for calc the hash ;)). It dosen't need a mutex there.

Reply:
signing is really super fast. but yeah i will make the change